src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -417,7 +417,7 @@ export const Deposit: React.FC<DepositProps> = ({
   }, [outboundFeeCryptoBaseUnit, assetPriceInFeeAsset, asset, feeAsset])
 
   const _validateCryptoAmount = useCallback(
-    async (value: string) => {
+    async (value: string): Promise<string | boolean | undefined> => {
```

Comment:
Love this and at the same time regarding the multiple types it returns it doesn't seems so useful to me, do we have a rule on where we decide to explicitely define the returns or not?

Reply:
We don't really have a convention, I believe the consensus is this is context-specific here!

Rationale on this specific one, was debugging things out there and this helped to ensure we were indeed returning what react-hook-form validation expects, but happy to remove it if that's verbose 🙏🏽

Reply:
> We don't really have a convention, I believe the consensus is this is context-specific here!
> 
> Rationale on this specific one, was debugging things out there and this helped to ensure we were indeed returning what react-hook-form validation expects, but happy to remove it if that's verbose 🙏🏽

If it helped out debugging an actual problem, we might need to explicitely set the return then and looks like we are fine by letting it in?

Reply:
Anyway, it's not blocking so feel free to take your own decision on it, I'm happy with both as long as we have an argument with or without

Reply:
0638463a07e5a41990773f8e7fc3593d20dee353